sextant/src/main.rs

```diff
@@ -66,9 +75,360 @@ enum Opt {
+        )
+    );
+
+    // copy arch specific root
+    log::debug!("copy arch specific root:");
+    let arch_dir = src_dir.join(format!("root-{}", arch));
+    log::debug!("arch_dir={}", arch_dir.as_os_str().to_str().unwrap());
+    if arch_dir.exists() {
+        let arc_spec_files = fs::read_dir(arch_dir)?
+            .map(|res| res.map(|e| e.path()))
+            .filter_map(Result::ok)
+            .collect::<Vec<PathBuf>>();
+        log::debug!("arc_spec_dirs.len()={}", arc_spec_files.len());
+        for arc_spec_file in arc_spec_files {
+            log::debug!(
+                "copy {} to {}",
+                arc_spec_file.as_os_str().to_str().unwrap(),
+                tmp_root_dir.as_os_str().to_str().unwrap()
+            );
+            // TODO: we assume copying a file and not a directory
+            std::fs::copy(
+                &arc_spec_file,
+                &tmp_root_dir.join(arc_spec_file.file_name().unwrap()),
+            )?;
+            // fs_extra::dir::copy(&arc_spec_file, &tmp_root_dir, &options)?;
+        }
+    }
+
+    println!("find {}:", tmp_dir.path().as_os_str().to_str().unwrap());
+    println!(
+        "{}",
+        String::from_utf8_lossy(
+            &Command::new("find")
+                .arg(tmp_dir.path().as_os_str().to_str().unwrap())
+                .output()?
+                .stdout
+        )
+    );
+
+    // write manifest
+    log::debug!("write manifest");
+    let tmp_manifest_dir = tmp_dir.path().join("manifest").with_extension("yaml");
+    log::debug!(
+        "create file {}",
+        tmp_manifest_dir.as_os_str().to_str().unwrap()
+    );
+    let tmp_manifest_file = File::create(&tmp_manifest_dir)?;
+    log::debug!(
+        "writing file {}",
+        tmp_manifest_dir.as_os_str().to_str().unwrap()
+    );
+    serde_yaml::to_writer(tmp_manifest_file, &manifest)?;
+
+    // remove existing containers
+    // TODO: remove all {registry}/#{name}-#{arch}-* directories
+
+    let npk_dir = registry_dir
+        .join(format!("{}-{}-{}", manifest.name, arch, manifest.version))
+        .with_extension("npk");
+    let fsimg_path = &tmp_dir.path().join("fs").with_extension("img");
+
+    /* The list of pseudo files is target specific.
+     * Add /lib and lib64 on Linux systems.
+     * Add /system on Android. */
+    let mut pseudo_files = vec![
+        ("/tmp", 444),
+        ("/proc", 444),
+        ("/dev", 444),
+        ("/sys", 444),
+        ("/data", 777),
+    ];
+    if arch == "aarch64-unknown-linux-gnu" || arch == "x86_64-unknown-linux-gnu" {
+        pseudo_files.push(("/lib", 444));
+        pseudo_files.push(("/lib64", 444));
+    } else if arch == "aarch64-linux-android" {
+        pseudo_files.push(("/system", 444));
+    }
+
+    // create filesystem image
+    let squashfs_comp = match std::env::consts::OS {
+        "linux" => "gzip",
+        _ => "zstd",
+    };
+    if fs_type == &FsType::SQUASHFS {
+        let mut cmd = Command::new("mksquashfs");
```

Comment:
The `mksquashfs` call shall be wrapped in a struct. Use the Builder pattern to add arguments like `.pseudo_file(t: PseudoType, p: &Path, permissions: ....)`